tasks/eth/006-MCP-L1/postCheck.s.sol

```diff
@@ -0,0 +1,231 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson} from "./SignFromJson.s.sol";
+import {ResourceMetering} from "@eth-optimism-bedrock/src/L1/ResourceMetering.sol";
+import {DeployConfig} from "@eth-optimism-bedrock/scripts/DeployConfig.s.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {Constants} from "@eth-optimism-bedrock/src/libraries/Constants.sol";
+import {L1StandardBridge} from "@eth-optimism-bedrock/src/L1/L1StandardBridge.sol";
+import {L2OutputOracle} from "@eth-optimism-bedrock/src/L1/L2OutputOracle.sol";
+import {ProtocolVersion, ProtocolVersions} from "@eth-optimism-bedrock/src/L1/ProtocolVersions.sol";
+import {SuperchainConfig} from "@eth-optimism-bedrock/src/L1/SuperchainConfig.sol";
+import {OptimismPortal} from "@eth-optimism-bedrock/src/L1/OptimismPortal.sol";
+import {L1CrossDomainMessenger} from "@eth-optimism-bedrock/src/L1/L1CrossDomainMessenger.sol";
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {Predeploys} from "@eth-optimism-bedrock/src/libraries/Predeploys.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {console2 as console} from "forge-std/console2.sol";
+
+contract PostCheck is SignFromJson {
+    Types.ContractSet prox = Types.ContractSet({
+        L1CrossDomainMessenger: 0x25ace71c97B33Cc4729CF772ae268934F7ab5fA1,
+        L1StandardBridge: 0x99C9fc46f92E8a1c0deC1b1747d010903E884bE1,
+        L2OutputOracle: 0xdfe97868233d1aa22e815a266982f2cf17685a27,
+        DisputeGameFactory: address(0),
+        OptimismMintableERC20Factory: 0x75505a97BD334E7BD3C476893285569C4136Fa0F,
+        OptimismPortal: 0xbEb5Fc579115071764c7423A4f12eDde41f106Ed,
+        OptimismPortal2: 0xbEb5Fc579115071764c7423A4f12eDde41f106Ed,
+        SystemConfig: 0x229047fed2591dbec1eF1118d64F7aF3dB9EB290,
+        L1ERC721Bridge: 0x5a7749f83b81B301cAb5f48EB8516B986DAef23D,
+        ProtocolVersions: 0x8062AbC286f5e7D9428a0Ccb9AbD71e50d93b935,
+        SuperchainConfig: 0x95703e0982140D16f8ebA6d158FccEde42f04a4C
```

Comment:
All of these addresses are in the superchain registry. I'd suggest adding the latest commit as a dependency via `forge install` and then reading in all addresses for mainnet that way

Reply:
do you know how to read a `.yaml` file from the solidity contract? `ProtocolVersions` and `SuperchainConfig` would be located in `lib/superchain-registry/superchain/configs/mainnet/superchain.yaml`

Reply:
There was a cheatcode for this, not sure if its been merged yet, otherwise will need some crazy bash with something like `yq`

Reply:
Toml reading/writing cheats were merged by not yaml. We have a root `just install` command that installs deps, so I'm ok with adding a new dep on yq or yj as long as we unsure the new install command is cross-platform (e.g. no `brew install`). The easiest way is probably do one ffi call that converts the yaml to json in a temp dir and returns the tmp filepath, and then just use the JSON cheats to read that file and delete it afterwards